wherehows-web/app/components/dataset-compliance-row.ts

```diff
@@ -0,0 +1,353 @@
+import DatasetTableRow from 'wherehows-web/components/dataset-table-row';
+import ComputedProperty, { alias } from '@ember/object/computed';
+import { computed, get, getProperties, getWithDefault } from '@ember/object';
+import {
+  Classification,
+  defaultFieldDataTypeClassification,
+  fieldIdentifierOptions,
+  fieldIdentifierTypeIds,
+  FieldIdValues,
+  hasPredefinedFieldFormat,
+  IComplianceField,
+  IFieldIdentifierOption,
+  isCustomId,
+  isMixedId,
+  logicalTypesForGeneric,
+  logicalTypesForIds,
+  SuggestionIntent,
+  IFieldFormatDropdownOption
+} from 'wherehows-web/constants';
+import { fieldChangeSetRequiresReview } from 'wherehows-web/utils/datasets/compliance-policy';
+import { isHighConfidenceSuggestion } from 'wherehows-web/utils/datasets/compliance-suggestions';
+import { hasEnumerableKeys } from 'wherehows-web/utils/object';
+
+export default class DatasetComplianceRow extends DatasetTableRow {
+  /**
+   * Declares the field property on a DatasetTableRow. Contains attributes for a compliance field record
+   * @type {IComplianceField}
+   * @memberof DatasetComplianceRow
+   */
+  field: IComplianceField;
+
+  /**
+   * Describes action interface for `onFieldIdentifierTypeChange`
+   * @memberof DatasetComplianceRow
+   */
+  onFieldIdentifierTypeChange: (field: IComplianceField, option: { value: FieldIdValues }) => void;
+
+  /**
+   * Describes action interface for `onFieldLogicalTypeChange`
+   * @memberof DatasetComplianceRow
+   */
+  onFieldLogicalTypeChange: (
+    field: IComplianceField,
+    option: { value: void } | Pick<IFieldFormatDropdownOption, 'value'>
+  ) => void;
+
+  /**
+   * Describes action interface for `onFieldClassificationChange`
+   * 
+   * @memberof DatasetComplianceRow
+   */
+  onFieldClassificationChange: (field: IComplianceField, option: { value: '' | Classification }) => void;
+
+  /**
+   * Describes action interface for `onSuggestionIntent`
+   * @memberof DatasetComplianceRow
+   */
+  onSuggestionIntent: (field: IComplianceField, intent?: SuggestionIntent) => void;
+
+  /**
+   * The field identifier attribute
+   * @type {ComputedProperty<string>}
+   * @memberof DatasetComplianceRow
+   */
+  identifierField: ComputedProperty<string> = alias('field.identifierField');
+
+  /**
+   * The field's dataType attribute
```

Comment:
Are those automated comments?

Reply:
not really, manually generated with document this in vscode, then modified
https://marketplace.visualstudio.com/items?itemName=joelday.docthis